parserator_web/static/js/index.js

```diff
@@ -1,2 +1,61 @@
-/* TODO: Flesh this out to connect the form to the API and render results
-   in the #address-results div. */
+/* eslint-disable strict */
+document.addEventListener('DOMContentLoaded', function () {
+  const addressForm = document.querySelector('.form')
+
+  addressForm.addEventListener('submit', async (e) => {
+    try {
+      const results = await parseAddressInput(e)
+      displayResults(results)
+    } catch (error) {
+      displayError(error.message)
+    }
+  })
+})
+
+const parseAddressInput = async (e) => {
+  e.preventDefault()
+  const input = document.getElementById('address').value
+  try {
+    const response = await fetch('/api/parse?input=' + encodeURIComponent(input))
+    if (!response.ok) {
+      const errorResponse = await response.json()
+      throw new Error(errorResponse.detail)
+    } else {
+      const jsonResponse = await response.json()
+      return jsonResponse
+    }
+  } catch (error) {
+    throw new Error(error.message)
+  }
+}
+
+const displayResults = (response) => {
+  const addressComponents = response.address_components
+  const addressType = response.address_type
+
+  document.getElementById('parse-type').innerHTML = addressType
+
+  const componentsToDisplay = document.getElementById('address-components')
+  componentsToDisplay.innerHTML = ''
+
+  let addressComponentsHTML = ''
+
+  for (const componentType in addressComponents) {
+    addressComponentsHTML += '<tr>'
+    addressComponentsHTML += '<td>' + addressComponents[componentType] + '</td>'
+    addressComponentsHTML += '<td>' + componentType + '</td>'
+    addressComponentsHTML += '</tr>'
+  }
```

Comment:
this may be place to use the methods of the [HTMLTableElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTableElement). 

what would be the advantages or disadvantages to that approach?

Reply:
Thanks for this suggestion! I had not used those before. 

It seems like, in this case, advantages would outweigh potential disadvantages, both because it would make it harder for small typos to introduce errors, and because I think it makes the code easier to follow and more secure. My sense is that the disadvantage of extra DOM manipulation might be a bigger issue if we expected the table to be very long, which isn't an issue here.